pkg/3scale/amp/operator/base_apimanager_logic_reconciler.go

```diff
@@ -65,31 +66,86 @@ func (r BaseAPIManagerLogicReconciler) ensureOwnerReference(obj common.Kubernete
 	return changed, nil
 }
 
-func (r *BaseAPIManagerLogicReconciler) createResource(obj common.KubernetesObject) error {
-	obj.SetNamespace(r.apiManager.GetNamespace())
-	if err := r.setOwnerReference(obj); err != nil {
-		return err
+func (r *BaseAPIManagerLogicReconciler) ReconcilePodDisruptionBudget(desired *v1beta1.PodDisruptionBudget, mutatefn reconcilers.MutateFn) error {
+	if !r.apiManager.IsPDBEnabled() {
+		common.TagObjectToDelete(desired)
 	}
+	return r.ReconcileResource(&v1beta1.PodDisruptionBudget{}, desired, mutatefn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileImagestream(desired *imagev1.ImageStream, mutatefn reconcilers.MutateFn) error {
```

Comment:
Maybe it would be a good idea to return the existing object on the reconcile methods?

The existing object might be useful in some situations outside the ReconcileXX method itself.

For example, in a Job object, we want to make sure the job is successfully created, and do something with it after that when exists. In this case we use the resulting existing object to do some stuff with it later:
```
	err := r.ReconcileResource(existing, desired, operator.CreateOnlyMutator)
	if err != nil {
	  return reconcile.Result{}, err
	}
	if existing.Status.Succeeded != *desired.Spec.Completions {
          ... // Do other stuff
        }
```

Reply:
Another usage is when you want to perform something just after creating the object. In that case existing and desired should be equal but only the spec part, not the status part. Using the returned 'status' section might be sometimes used and that is not included in the desired object even if the object is being created for the first time

Reply:
I think that status will be eventually updated and you will not have this info in the "existing" object. You will need to read it from the cluster again to be sure. 

Anyway, `BaseAPIManagerLogicReconciler.ReconcileImagestream` does not need to return existing object in anyway.

If you ever need existing object after reconciliation, using the base method, 
```
r.BaseReconciler.ReconcileResource(existingObj, desired, myMutator)
```
the existing object, after the call, will be the result of the reconciliation.

Reply:
> I think that status will be eventually updated and you will not have this info in the "existing" object. You will need to read it from the cluster again to be sure.

That depends on the object type. It's often the case you say. For some object types use the returned object in the same reconciliation loop iteration it's often useful and you might have the information already there.

You can avoid performing a Get just after performing a ReconcileImageStream call in the same reconcile loop call. You also will avoid eventual consistency caching in that iteration etc... . That's one of the reasons why the create method returns the object.

Reply:
In any case I guess we can leave it this way and if we see we often resort to the situation I explained we can reevaluate it

Reply:
:+1: As I said, you can always use `r.BaseReconciler.ReconcileResource(existingObj, desired, myMutator)`, and there you have required info.